# Instrucciones

1. Determine el modelo de regresión lineal que le permita entender la relación entre la variable Gastos y las variables regresoras Años_capacitación, Años_Cargo, Años_Compañía	y Edad.
2. ¿Qué puede concluir respecto de los estadísticos globales del modelo?
3. ¿Qué puede concluir respecto de los estadísticos individuales del modelo?

In [ ]:
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.formula.api as smf # Nos permite ajustar el modelo de regresión

In [ ]:
# Importamos la base de datos y la denominamos df por "data frame"
df = pd.read_csv('solemne_forma_2.csv', sep=';', decimal=",")

In [ ]:
df

Gastos  Años_capacitación  Años_Cargo  Años_Compañía       Edad
0    1.321502e+06                 11           1              2  54.250000
1    9.635870e+05                 19           1              1  54.568493
2    1.000000e-29                  0           1              3  33.263699
3    2.506052e+06                 13           1              3  27.063014
4    4.082858e+06                  0           1              3  61.896347
..            ...                ...         ...            ...        ...
980  7.191680e+05                 22           1              2  28.957763
981  8.006480e+05                 19           1              2  28.960502
982  2.604411e+06                 11           1              3  36.194064
983  5.367600e+05                 24           1              1  24.521918
984  1.704063e+06                 14           1              3  26.196804

[985 rows x 5 columns]

In [ ]:
df.describe()

Gastos  Años_capacitación  Años_Cargo  Años_Compañía        Edad
count  9.850000e+02         985.000000  985.000000     985.000000  985.000000
mean   1.556273e+06          12.591878    1.187817       2.646701   41.955940
std    1.054048e+06           8.310167    0.496178       0.749696   12.484696
min    1.000000e-29           0.000000    1.000000       1.000000   21.573973
25%    7.538530e+05           6.000000    1.000000       2.000000   31.266438
50%    1.264120e+06          14.000000    1.000000       3.000000   38.855251
75%    2.147296e+06          21.000000    1.000000       3.000000   53.288356
max    6.116213e+06          24.000000    5.000000       4.000000   73.879909

# Solución

1.

In [ ]:
# Usamos el método de los mínimos cuadrados para ajustar la regresión lineal multiple
Modelo = smf.ols('Gastos ~ Años_capacitación+Años_Cargo+Años_Compañía+Edad', df).fit() #Ajustamos el modelo
# Usamos summary para pedir el resultado del modelo
Modelo.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                 Gastos   R-squared:                       0.482
Model:                            OLS   Adj. R-squared:                  0.480
Method:                 Least Squares   F-statistic:                     228.1
Date:                Thu, 18 Mar 2021   Prob (F-statistic):          2.23e-138
Time:                        02:39:50   Log-Likelihood:                -14733.
No. Observations:                 985   AIC:                         2.948e+04
Df Residuals:                     980   BIC:                         2.950e+04
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
Intercept          2.003e+06   1.88e+05     10.644      0.000    1.63e+06    2.37e+06
Años_capacitación -7.424e+04   3542.683    -20.957      0.000   -8.12e+04   -6.73e+04
Años_Cargo        -3.728e+05   5.05e+04     -7.377      0.000   -4.72e+05   -2.74e+05
Años_Compañía      8.212e+04   3.59e+04      2.290      0.022    1.18e+04    1.52e+05
Edad                 1.7e+04   2097.642      8.103      0.000    1.29e+04    2.11e+04
==============================================================================
Omnibus:                       78.845   Durbin-Watson:                   1.969
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              216.250
Skew:                           0.410   Prob(JB):                     1.10e-47
Kurtosis:                       5.144   Cond. No.                         359.
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

1. El modelo obtenido es $Gastos=2.003exp^{6}-7.424exp^{4}* Años\_capacitación	-3.728exp^{5}*Años\_Cargo+	8.212exp^{4}*Años\_Compañía+1.7exp^{4}*Edad$

2. Análisis estadísticos globales. A partir de los valores de F, R2 y R2 Adj podemos concluir que el modelo es capaz de explicar cerca del 48% de la varianza. F es significativo ($F_{est} < pvalor$). Luego, se rechaza la hipótesis nula de que los coeficientes de regresiones son iguales a cero, por lo que al menos un coeficiente debe ser distinto de cero.

3. Los coeficientes asociados a cada variable mantienen un test-t estadísticamente significativo ($ t_{est} < pvalor$), por lo que se rechaza la hipótesis nula para estas variables. Es decir, su coeficiente de regresión es distinto a cero.
